### PR TITLE
Use proper exception in lostController

### PR DESCRIPTION
**Pull request by rullzer**
There is no need to log the expcetion of most of the stuff here.
We should properly log them but an exception is excessive.

This moves it to a proper exception which we can catch and then log.
The other exceptions will still be fully logged.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: core/Controller/LostController.php [hunk 1]
```diff
@@ -248,11 +249,11 @@ public function email($user){
 		// FIXME: use HTTP error codes
 		try {
 			$this->sendEmail($user);
-		} catch (\Exception $e) {
+		} catch (ResetPasswordException $e) {
 			// Ignore the error since we do not want to leak this info
-			$this->logger->logException($e, [
-				'level' => ILogger::WARN
-			]);
+			$this->logger->warning($e->getMessage());
```
Comment (MorrisJobke) [inline core/Controller/LostController.php]: Giving it more context than just a log entry with "user disabled" might be helpful. Maybe add "Could not send password reset email: $message"?

Comment (MorrisJobke) [reply core/Controller/LostController.php]:
```suggestion
			$this->logger->warning('Could not send password reset email: ' . $e->getMessage());
```

File: core/Controller/LostController.php [hunk 2]
```diff
@@ -379,14 +377,10 @@ protected function sendEmail($input) {
 	 * @throws \InvalidArgumentException
```
Comment (kesselb) [inline core/Controller/LostController.php]:
```suggestion
	 * @throws \InvalidArgumentException
	 * @throws ResetPasswordException
```

File: core/Controller/LostController.php [hunk 3]
```diff
@@ -319,9 +320,7 @@ protected function sendEmail($input) {
 		$email = $user->getEMailAddress();
 
 		if (empty($email)) {
-			throw new \Exception(
-				$this->l10n->t('Could not send reset email because there is no email address for this username. Please contact your administrator.')
-			);
+			throw new ResetPasswordException('Could not send reset e-mail since there is no email for username ' . $input);
```
Comment (kesselb) [inline core/Controller/LostController.php]: Please update the phpdoc for sendEmail method
```
	 * @throws ResetPasswordException
	 * @throws \OCP\PreConditionNotMetException
```

**Comment by rullzer** [comment]
Done and done

File: core/Controller/LostController.php [hunk 4]
```diff
@@ -312,16 +313,14 @@ public function setPassword($token, $userId, $password, $proceed) {
 
 	/**
 	 * @param string $input
-	 * @throws \Exception
+	 * @throws ResetPasswordException
```
Comment (kesselb) [inline core/Controller/LostController.php]:
```suggestion
	 * @throws ResetPasswordException
	 * @throws \OCP\PreConditionNotMetException
```

`setUserValue` can throw a `PreConditionNotMetException`.

**Review by kesselb — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**